### PR TITLE
Various clean ups - including defaulting to webgl on web 

### PR DESCRIPTION
**Pull request by ry**

### CODE REVIEW

File: src/dl/math/backends/webgl/select_gpu.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  constructor(shape: number[]) {
+    this.outputShape = shape.slice();
+    const dType = getCoordsDataType(shape.length);
+    const sampleCoords = getSampleCoords(shape.length);
+
+    this.userCode = `
+      void main() {
+        ${dType} coords = getOutputCoords();
+        int cond = int(getCond(${sampleCoords}));
```
Comment (piscisaureus) [inline src/dl/math/backends/webgl/select_gpu.ts]: just for my understanding - where does getCond come from?

File: src/tensor_util.ts [hunk 2]
```diff
@@ -211,3 +211,7 @@ export function inferShape(arr: number | boolean | RegularArray<number> |
   }
   return shape;
 }
+
+export function isTensor(t: any): t is Tensor {
+  return t.id && !!t.dtype && !!t.device;
```
Comment (piscisaureus) [inline src/tensor_util.ts]: Why not (t instanceof Tensor) ?

File: src/tensor.ts [hunk 3]
```diff
@@ -24,7 +24,14 @@ import { assert, assertEqual, IS_NODE } from "./util";
 
 export function convert(t: types.TensorLike,
                         opts?: types.TensorOpts): Tensor {
-  if (t instanceof Tensor) return t;
+  if (t instanceof Tensor) {
```
Comment (piscisaureus) [inline src/tensor.ts]: Why do you use `isTensor` elsewhere, but `instanceof Tensor` here?

File: src/tensor.ts [hunk 4]
```diff
@@ -22,9 +22,27 @@ import { allFinite, assertShapesEqual } from "./tensor_util";
...
+ *
+ * If a tensor is given to tensor, it simply returns it.
+ *
+ * @arg t - An array of numbers, representing a Tensor. Or a Tensor in which
+ *          case tensor() acts as the identity function.
+ * @arg args - An object like this { dtype: "int32", device: "GPU:0" }
+ */
+export function tensor(t: types.TensorLike,
+                       opts?: types.TensorOpts): Tensor {
+  if (t instanceof Tensor) {
```
Comment (piscisaureus) [inline src/tensor.ts]: Why do you use `instanceof Tensor` here, and `isTensor` elsewhere?

File: src/dl/math/backends/webgl/select_gpu.ts (hunk 1 above)
Comment (ry) [reply src/dl/math/backends/webgl/select_gpu.ts]: https://github.com/propelml/propel/blob/38597c77f9d2bfc0baf2853c5d9aa54b9a78d231/src/dl/math/backends/webgl/shader_compiler.ts#L422-L430

File: src/tensor_util.ts (hunk 2 above)
Comment (ry) [reply src/tensor_util.ts]: Oh, I guess that works. Fixed.

File: src/tensor.ts (hunk 4 above)
Comment (ry) [reply src/tensor.ts]: Fixed.

**Pull request merged**